module/apps/actor-armor.js

```diff
@@ -65,31 +64,28 @@ export default class ActorArmorConfig extends DocumentSheet {
   async _onChangeInput(event) {
     await super._onChangeInput(event);
 
-    // Reference actor data
-    let actorData = this.object.toObject(false);
-    let ac = actorData.system.attributes.ac;
+    const ac = this.clone.system.attributes.ac;
 
     // Reference form data
-    const calc = this.form["ac.calc"].value;
-    const cfg = CONFIG.DND5E.armorClasses[calc];
-    const enableFlat = ["flat", "natural"].includes(calc);
+    ac.calc = this.form["ac.calc"].value;
+    const cfg = CONFIG.DND5E.armorClasses[ac.calc];
+    const enableFlat = ["flat", "natural"].includes(ac.calc);
 
     // Handle changes to the calculation mode specifically
-    let formula = this.form["ac.formula"].value;
+    ac.formula = this.form["ac.formula"].value;
     let flat = this.form["ac.flat"].value;
     if ( event.currentTarget.name === "ac.calc" ) {
-      formula = calc === "custom" ? ac.formula : cfg.formula;
+      ac.formula = ac.calc === "custom" ? ac.formula : cfg.formula;
       if ( enableFlat ) flat = ac.flat;
     }
 
     // Recompute effective AC
-    actorData = foundry.utils.mergeObject(actorData, {"system.attributes.ac": {calc, formula}});
-    if ( enableFlat ) actorData.system.attributes.ac.flat = flat;
-    ac = this.object._computeArmorClass(actorData.data);
+    if ( enableFlat ) ac.flat = flat;
+    this.clone._prepareArmorClass();
```

Comment:
Rather than modifying the actor data directly and then calling `_prepareArmorClass` it should be better to instead compute an object of `changes` to apply to the clone and then call `this.clone.updateSource(changes)` - AC preparation for the clone will happen automatically.